src/oss/langchain-models.mdx

```diff
@@ -2,14 +2,605 @@
+    "Why do parrots have colorful feathers?",
+    "How do airplanes fly?",
+    "What is quantum computing?"
+])
+for response in responses:
+    print(response)
+```
+
+By default, `batch()` will only return the final output for the entire batch.
+If you want to receive the output for each individual input as it is finishes
+generating, you can stream results with `batch_as_completed()`:
+
+```python Yield responses upon completion
+for response in model.batch_as_completed([
+    "Why do parrots have colorful feathers?",
+    "How do airplanes fly?",
+    "What is quantum computing?"
+]):
+    print(response)
+```
+<Note>
+    When using `batch_as_completed()`, results may arrive out of order. Each
+    includes the input index for matching to reconstruct the original order if
+    needed.
+</Note>
+
+For more details on batching, see the [batching]() reference.
+
+## Advanced features
+
+### Tool calling
+
+Models can request to call tools that perform tasks such as fetching data from a
+database, searching the web, or running code.
+
+<Note>
+    You will sometimes hear the term `function calling`. We use this term 
+    interchangeably with `tool calling`.
+</Note>
+
+To make tools that you have defined available for use by a model, you must bind
+them using `.bind_tools()`. In subsequent invocations, the model can choose to
+call any of the bound tools as needed.
+
+Some model providers offer built-in tools that can be enabled via model
+parameters. In such cases, you do not need to define or bind those tools to your
+model. Check the respective [provider reference]() for details.
+
+<CodeGroup>
+    ```python Binding user tools highlight={7}
+    from pydantic import BaseModel, Field
+
+    class GetWeather(BaseModel):
+        """Get the current weather in a given location"""
+        location: str = Field(..., description="The city and state, e.g. Boston, MA")
+
+    model_with_tools = model.bind_tools([GetWeather])
+
+    response = model_with_tools.invoke("What's the weather like in Boston?")
+    for tool_call in response.tool_calls:
+        # View tool calls made by the model
+        print(f"Tool: {tool_call['name']}")
+        print(f"Args: {tool_call['args']}")
+    ```
+    
+    ```python Built-in tool usage with citations
+    response = model.invoke("Search the web for the latest news on AI.")
+    content = response.content_blocks
+
+    # content: [
+    #     TextContentBlock(
+    #         text = "The latest news articles on AI..."),
+    #         annotations = [
+    #             Citation(
+    #                 title="AI breakthrough in natural language processing",
+    #                 url="https://example.com/ai-news",
+    #                 cited_text="Researchers have...",
+    #         )]
+    # ])
+    ```
+</CodeGroup>
+
+For more details on creating, using, and executing tools requests, see the
+[tools]() guide.
+
+{/* 
+
+If we want to include some tool info here, we can do so like this:
```

Comment:
yes! all this stuff is good. things like defining tools will be in tools page, but other than that is largely about using tools in model, not agent. using tools in model should go here